src/libxtp/gwbse/bse.cc

```diff
@@ -44,9 +44,51 @@ void BSE::configure(const options& opt,
   _bse_vtotal = _bse_vmax - _opt.vmin + 1;
   _bse_ctotal = _opt.cmax - _bse_cmin + 1;
   _bse_size = _bse_vtotal * _bse_ctotal;
+  _Hqp = Scale_or_Expand_HQP(RPAInputEnergies);
   SetupDirectInteractionOperator(RPAInputEnergies);
 }
 
+Eigen::MatrixXd BSE::Scale_or_Expand_HQP(
```

Comment:
Looks much nicer but I do not get the scale part of the name?

Reply:
How about Cut_or_Expand?
Actually it will do both, if required at the same time. Like if bsemin > qpmin but bsemax > qpmax, it will "cut" for min and expand for max. Maybe CutExpand?

Reply:
AdjustHqpSize?

Reply:
Done